components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -853,6 +854,27 @@ protected void doAuthentication(HttpServletRequest request, HttpServletResponse
             // store authenticated idp
             stepConfig.setAuthenticatedIdP(idpName);
             authenticatedIdPData.setIdpName(idpName);
+
+            // This fix needs to be generalized and improved for non organization login flows as well.
+            if (isLoggedInWithOrganizationLogin(authenticatorConfig)) {
+                String orgId = authenticatedUser.getUserResidentOrganization();
+                List<AuthenticatorConfig> authenticators = authenticatedIdPData.getAuthenticators();
+                Iterator<AuthenticatorConfig> iterator = authenticators.iterator();
+                while (iterator.hasNext()) {
+                    AuthenticatorConfig config = iterator.next();
+                    if (config.getParameterMap() != null && StringUtils.equals(orgId,
+                            config.getParameterMap().get(FrameworkConstants.ORG_ID_PARAMETER))) {
+                        iterator.remove();
+                        authenticatedIdPData.setAuthenticators(authenticators);
+                        break;
+                    }
+                }
+                if (authenticatorConfig.getParameterMap() == null) {
+                    authenticatorConfig.setParameterMap(new HashMap<>());
+                }
+                authenticatorConfig.getParameterMap().put(FrameworkConstants.ORG_ID_PARAMETER, orgId);
```

Comment:
Shall we move this to a private method.

Reply:
(orgId, authenticatedIdPData)

Reply:
+1. WIll do the change.

Reply:
Updated with [99a44c1](https://github.com/wso2/carbon-identity-framework/pull/6846/commits/99a44c1b89aa0f1fc94be83240f8e2888102fd6c)

Additionally, added isLoggedInWithOrganizationLogin(config) check before the removal